src/components/typography/text/text.js

```diff
@@ -98,12 +203,30 @@ const Body = props =>
       {props.children}
     </p>
   );
+};
 
 Body.displayName = 'TextBody';
 Body.propTypes = {
+  as: PropTypes.oneOf(['span', 'p']),
   isBold: PropTypes.bool,
   isItalic: PropTypes.bool,
-  isInline: PropTypes.bool,
+  isInline(props, propName, componentName, ...rest) {
+    if (!isNil(props[propName])) {
+      const message = oneLine`
+            "${propName}" property of "${componentName}" has been deprecated
+            and will be removed in the next major version.\n
+            Please use "as" prop instead.`;
+      warning(false, message);
+
+      if (props.as) {
+        return new Error(oneLine`
+        Invalid prop "${propName}" supplied to "${componentName}".
+        "${propName}" does not have any effect when "as" is defined`);
+      }
+    }
+
+    return PropTypes.bool(props, propName, componentName, ...rest);
```

Comment:
Idea. Maybe we can share these prop-types in `typography/prop-types`. I know that they will be removed but it makes the actual code less noisy.